scripts/apps/authoring/authoring/directives/VideoTimeline.ts

```diff
@@ -0,0 +1,317 @@
+                    TweenMax.set(maskright, {
+                        width: ((1 - position) * 100) + '%'
+                    });
+                }
+            };
+
+            function getPositionBar(pX) {
+                var position = ((pX - controlbar.getBoundingClientRect().left) / controlbar.offsetWidth);
+                if (position > 1) {
+                    position = 1;
+                }
+                if (position < 0) {
+                    position = 0;
+                }
+                position = Math.floor(position * 100) / 100;
+                return position;
+            };
+
+            function getstrtime(s) {
+                mins = Math.floor(s / 60);
+                mins = mins < 10 ? '0' + mins : mins;
+                secs = Math.floor(s % 60);
+                secs = secs < 10 ? '0' + secs : secs;
+                li = Math.floor((s * 10) % 10);
+                return mins + ':' + secs + '.' + li;
+            };
+
+            function onDragCb(type) {
+                var position = getPositionBar(PositionX);
+                if (type == 'right') {
+                    TweenMax.set(cbwrapper, {
+                        right: ((1 - position) * 100) + '%'
+                    });
+                    TweenMax.set(maskright, {
+                        width: ((1 - position) * 100) + '%'
+                    });
+                    barright.setAttribute("data-content", getstrtime(position * scope.video.duration));
+                    scope.cut.end = position * scope.video.duration;
+                } else {
+                    TweenMax.set(cbwrapper, {
+                        left: (position * 100) + '%'
+                    });
+                    TweenMax.set(maskleft, {
+                        width: (position * 100) + '%'
+                    });
+                    barleft.setAttribute("data-content", getstrtime(position * scope.video.duration));
+                    scope.cut.start = position * scope.video.duration;
+                }
+            };
+
+
+            function onDragEndCb() {
+                setTimeline();
+                scope.onChange();
+            };
+
+            function onDragStart() {
+                var img = document.createElement("img");
+                event.dataTransfer.setDragImage(img, 0, 0);
+                event.dataTransfer.setData('text/plain', null)
+            };
+
+
+            function setTimeline() {
+                var position = getPositionBar(event.clientX);
+                scope.video.currentTime = position * scope.video.duration;
+                inner.innerHTML = getstrtime(scope.video.currentTime);
+                TweenMax.set(progressoutput, {
+                    left: (position * 100) + '%'
+                });
+                TweenMax.set(progressoutput, {
+                    left: (position * 100) + '%'
+                });
+                return position;
+            };
+
+            function getRandomSpan() {
```

Comment:
After edit video(crop, or rotate), the data of these frames in server is change. But the `url` get these list frames not change, that mean it will get old cache data. So we use a random tag number for avoid the old cache data and reload new list frame for new edit video.